server/src/main/java/org/elasticsearch/search/aggregations/support/ValuesSourceRegistry.java

```diff
@@ -139,6 +195,21 @@ public AggregatorSupplier getAggregator(ValuesSourceConfig valuesSourceConfig, S
         throw  new AggregationExecutionException("Unregistered Aggregation [" + aggregationName + "]");
     }
 
+    public <T extends CompositeSupplier> T getComposite(Class<T> supplierClass, ValuesSourceConfig config) {
+        if (supplierClass != null && compositeRegistry.containsKey(supplierClass)) {
+            CompositeSupplier supplier = compositeRegistry.get(supplierClass).get(config.valueSourceType());
+            if (supplier == null) {
+                // TODO: Fix this error message! The class name of the supplier isn't going to mean anything to a user
+                throw new IllegalArgumentException(
+                    config.getDescription() + " is not supported for composite source [" + supplierClass.getCanonicalName() + "]"
```

Comment:
@nik9000 In this model, we no longer have convenient access to the composite key type name here, so we can't throw a well formed error message.  I kind of feel like this means we should be using the `CompositeValuesSourceBuilder` class as the key here, since we can at least use reflection to get a name out of that.  I don't love it though.  Thoughts?

Reply:
I see what you mean. I guess you could put the nice name on the `supplierClass` somehow. That seems like overkill though.

By using the builder as the key do you mean that it'd have some `Class<? extends CompositeSupplier> supplierClass` method on it? Maybe that isn't too bad? Or maybe we could just pass the name of the supplier in here and declare it as "for error messages".